lib/round-tracker.js

```diff
@@ -0,0 +1,78 @@
+import { createMeridianContract } from './meridian/contract.js'
+
+export async function createRoundGetter (pgClient) {
+  const contract = await createMeridianContract()
+
+  let sparkRound
+  const updateSparkRound = async (meridianRound) => {
+    sparkRound = await mapCurrentMeridianRoundToSparkRound({
+      meridianContractAddress: contract.address,
+      meridianRound: BigInt(meridianRound),
+      pgClient
+    })
+    console.log('SPARK round started: %s', sparkRound)
+  }
+
+  contract.on('RoundStart', updateSparkRound)
+  updateSparkRound(await contract.currentRoundIndex())
+
+  return () => sparkRound
+}
+
+export async function mapCurrentMeridianRoundToSparkRound ({
+  meridianContractAddress,
+  meridianRound,
+  pgClient
+}) {
+  let sparkRound
+
+  const { rows: [lastMeridianRecord] } = await pgClient.query(
+    'SELECT * FROM meridian_contract_versions ORDER BY last_spark_round DESC LIMIT 1'
```

Comment:
We could also query the new `spark_rounds` table for this, does any option have an advantage?

Reply:
We need to find the `contract_address` we used for the previous SPARK round. That information is not stored in the `spark_rounds` table.